lib/DxilDia/DxcPixLiveVariables.cpp

```diff
@@ -29,6 +29,35 @@
 
 #include <unordered_map>
 
+namespace std {
+template <> struct hash<std::pair<llvm::DIScope *, llvm::DIScope *>> {
+  size_t
+  operator()(std::pair<llvm::DIScope *, llvm::DIScope *> const &p) const {
+    const uint64_t f = reinterpret_cast<uint64_t>(p.first);
+    const uint64_t s = reinterpret_cast<uint64_t>(p.second);
+    std::hash<uint64_t> h;
+    return (h(f) ^ h(s));
+  }
+};
+} // namespace std
+
+// If a function is inlined, then the scope of variables within that function
+// will be that scope. Since many such callers may inline the function, we
+// actually need a "scope" that's unique to each "instance" of that function.
+// The caller's scope would be unique, but would have the undesirable
+// side-effect of smushing all of the function's variables together with the
+// caller's variables, at least from the point of view of PIX. The pair of
+// scopes (the function's and the caller's) is both unique to that particular
+// inlining, and distinct from the caller's scope by itself.
+static std::pair<llvm::DIScope *, llvm::DIScope *>
+GetUniqueScopeForPossiblyInlinedFunction(llvm::DebugLoc const &DbgLoc,
+                                         llvm::DIScope *VariableScope) {
+  llvm::DIScope *inlinedScope =
+      llvm::dyn_cast<llvm::DIScope>(DbgLoc.getInlinedAtScope());
```

Comment:
I think this better be `dyn_cast_or_null`, because `DbgLoc` is not guaranteed to be inlined.